crane/unit_of_work.go

```diff
@@ -0,0 +1,239 @@
+	}
+	if len(args) > 1 {
+		executeCommand("docker", args)
+	} else {
+		printErrorf("None of the targeted container is running.\n")
+	}
+}
+
+func (uow *UnitOfWork) Status(noTrunc bool) {
+	uow.Targeted().Status(noTrunc)
+}
+
+// Push containers.
+func (uow *UnitOfWork) Push() {
+	for _, container := range uow.Targeted() {
+		container.Push()
+	}
+}
+
+// Unpause containers.
+func (uow *UnitOfWork) Unpause() {
+	for _, container := range uow.Targeted() {
+		container.Unpause()
+	}
+}
+
+// Pause containers.
+func (uow *UnitOfWork) Pause() {
+	for _, container := range uow.Targeted().Reversed() {
+		container.Pause()
+	}
+}
+
+// Start containers.
+func (uow *UnitOfWork) Start() {
+	for _, container := range uow.Containers() {
+		if includes(uow.targeted, container.Name()) {
+			container.Start(excluded, cfg.Path())
+		} else if includes(uow.requireStarted, container.Name()) || !container.Exists() {
+			container.Start(excluded, cfg.Path())
+		}
+	}
+}
+
+// Stop containers.
+func (uow *UnitOfWork) Stop() {
+	for _, container := range uow.Targeted().Reversed() {
+		container.Stop()
+	}
+}
+
+// Kill containers.
+func (uow *UnitOfWork) Kill() {
+	for _, container := range uow.Targeted().Reversed() {
+		container.Kill()
+	}
+}
+
+func (uow *UnitOfWork) Exec(cmds []string) {
+	for _, container := range uow.Containers() {
+		if includes(uow.targeted, container.Name()) {
+			container.Exec(cmds, cfg.Path())
+		} else if includes(uow.requireStarted, container.Name()) || !container.Exists() {
+			container.Start(excluded, cfg.Path())
+		}
+	}
+}
+
+// Rm containers.
+func (uow *UnitOfWork) Rm(force bool) {
+	for _, container := range uow.Targeted().Reversed() {
+		container.Rm(force)
+	}
+}
+
+// Create containers.
+func (uow *UnitOfWork) Create(cmds []string, excluded []string) {
+	for _, container := range uow.Containers() {
+		if includes(uow.targeted, container.Name()) {
+			container.Create(cmds, excluded, cfg.Path())
+		} else if includes(uow.requireStarted, container.Name()) || !container.Exists() {
+			container.Start(excluded, cfg.Path())
+		}
+	}
+}
+
+// Provision containers.
+func (uow *UnitOfWork) Provision(noCache bool) {
+	uow.Targeted().Provision(noCache)
+}
+
+// Pull containers.
+func (uow *UnitOfWork) PullImage() {
+	for _, container := range uow.Targeted() {
+		if len(container.Dockerfile()) == 0 {
+			container.PullImage()
+		}
+	}
+}
+
+// Log containers.
+func (uow *UnitOfWork) Logs(follow bool, timestamps bool, tail string, colorize bool, since string) {
+	uow.Targeted().Logs(follow, timestamps, tail, colorize, since)
+}
+
+func (uow *UnitOfWork) Containers() Containers {
+	c := []Container{}
+	for _, name := range uow.order {
+		c = append(c, cfg.Container(name))
+	}
+	return c
+}
+
+func (uow *UnitOfWork) Targeted() Containers {
+	c := []Container{}
+	for _, name := range uow.order {
+		if includes(uow.targeted, name) {
+			c = append(c, cfg.Container(name))
+		}
+	}
+	return c
+}
+
+func (uow *UnitOfWork) Affected() []string {
```

Comment:
`Affected()` has a completely different meaning when it comes to targeting/cascading. Shouldn't it be `Required()` or `Dependencies()`?